### PR TITLE
Adding extensible enum tests

### PR DESCRIPTION
**Pull request by dsgouda**

### CODE REVIEW

**Comment by olydis** [comment]
# 🤖 AutoRest automatic publish job 🤖
~~~ Haskell
> will publish once PR gets merged
~~~

File: swagger/extensible-enums-swagger.json [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+      "Saturday",
+      "Sunday"
+     ],
+     "x-ms-enum": {
+      "name": "DaysOfWeekExtensibleEnum",
+      "modelAsString": true
+     },
+     "default": "Friday"
+    },
+    "IntEnum": {
```
Comment (olydis) [inline swagger/extensible-enums-swagger.json]: how would this compare to "simply" modelling it as
``` JSON
"enum": [
 "1",
 "1.1",
 "1.2",
 "1.3",
 "2",
 "2.1",
 "2.2",
 "3",
 "3.1",
 "3.3"
],
"x-ms-enum": {
 "modelAsString": true,
 "name": "IntEnum",
 "values": [
  {
   "value": "1",
   "description": "one",
   "name": "1"
  },
  {
   "value": "1.1",
   "description": "one",
   "name": "1"
  },
  {
   "value": "1.2",
   "description": "one",
   "name": "1"
  },
  {
   "value": "1.3",
   "description": "one",
   "name": "1"
  },
  {
   "value": "2",
   "description": "two",
   "name": "2"
  },
  {
   "value": "2.1",
   "description": "two",
   "name": "2"
  },
  {
   "value": "2.2",
   "description": "two",
   "name": "2"
  },
  {
   "value": "3",
   "description": "three",
   "name": "3"
  },
  {
   "value": "3.1",
   "description": "three",
   "name": "3"
  },
  {
   "value": "3.3",
   "description": "three",
   "name": "3"
  }
 ]
}
```
It's slightly more verbose but:
- uses the syntax we already have (which supports mapping multiple values to the same name!)
- the bare Swagger doesn't lie (the alternative values may appear on the wire, so they should appear in the `enum`)

**Comment by olydis** [comment]
don't forget to bump the version, otherwise kewl

**Comment by dsgouda** [comment]
Discussed this with @fearthecowboy and we'd like to maintain a link between what are the allowed values and what base value they map to, which requires we go by the `allowedValues` array we've here

**Comment by olydis** [comment]
# 🤖 AutoRest automatic publish job 🤖
~~~ Haskell
> will publish once PR gets merged
~~~

**Comment by olydis** [comment]
# 🤖 AutoRest automatic publish job 🤖
~~~ Haskell
> will publish once PR gets merged
~~~

File: routes/extensibleEnums.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        }
+        else{
+            utils.send400(res, next, 'Pet not found for '+petId);
+        }
+    });
+
+    router.post('/pet/addPet', function(req, res, next) {
+        var petName = req.body.name;
+        console.log('Inside addPet for '+petName+'\n');
+        if (petName === 'Retriever') {
```
Comment (olydis) [inline routes/extensibleEnums.js]: Why the stateful implementation via `retriever`? Why not simply reflect back the request (and maybe check that the request contains DaysOfWeek and IntEnum, to ensure it's not boring)?

Comment (dsgouda) [reply routes/extensibleEnums.js]: The other 2 tests cover that, this was with the intention of pure roundtripping.

Comment (olydis) [reply routes/extensibleEnums.js]: huh? my point is: this test isn't purely round tripping. Is uses some object outside of the function (why?), manipulates it, sends it back - why not literally round-trip?

Comment (dsgouda) [reply routes/extensibleEnums.js]: Ahh yes, let me fix that asap

**Pull request merged**